include/boost/gil/detail/math.hpp

```diff
@@ -10,15 +11,29 @@
 
 #include <array>
 #include <boost/gil/extension/numeric/kernel.hpp>
+#include <vector>
 
 namespace boost { namespace gil { namespace detail {
 
+enum class kernel_type
+{
+    sobeldx,
+    sobeldy,
+};
+
 static constexpr double pi = 3.14159265358979323846;
 
 static constexpr std::array<float, 9> dx_sobel = {{-1, 0, 1, -2, 0, 2, -1, 0, 1}};
+static constexpr std::array<float, 25> dx_sobel2 = {{
+    -1,-2,0,2,1,-4,-8,0,8,4,-6,-12,0,12,6,-4,-8,0,8,4,-1,-2,0,2,1
+}}; 
 static constexpr std::array<float, 9> dx_scharr = {{-1, 0, 1, -1, 0, 1, -1, 0, 1}};
 static constexpr std::array<float, 9> dy_sobel = {{1, 2, 1, 0, 0, 0, -1, -2, -1}};
+static constexpr std::array<float, 25> dy_sobel2 = {{
```

Comment:
What does `2` stand for in `sobel2`?

Reply:
```2``` in ```dy_sobel2``` indicates that the order of Sobel derivative in y-direction is 2, similar terminology is used for ```dx_sobel2```. I have added comment for clarifying this to future readers, is this resolvable now?

Reply:
Good